### PR TITLE
ci: Transfer ci into github action

### PR DESCRIPTION
**Pull request by Prnyself**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Override automatic language detection by changing the below list
+        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
+        language: ['go']
```
Comment (Xuanwo) [inline .github/workflows/codeql-analysis.yml:27]: CodeQL doesn't support ruby for now.

File: .github/workflows/unit-test.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        ruby: [ "2.5", "2.6", "2.7" ]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    steps:
+      - name: Set up ruby >2.5
+        uses: actions/setup-ruby@v1.1.2
```
Comment (Xuanwo) [inline .github/workflows/unit-test.yml:17]:
```suggestion
        uses: actions/setup-ruby@v1
```

@v1 is enough here.

Comment (Prnyself) [reply .github/workflows/unit-test.yml]: use ruby/setup-ruby instead

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (Prnyself) [reply .github/workflows/codeql-analysis.yml]: removed

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**